Source/Common/Robot.py

```diff
@@ -0,0 +1,105 @@
+# This class will represent the robot in the simulation
+# It will only handle information about the robot, not the actual drawing of the robot
+# that will be handled by the RobotDrawer class
+
+# the robot is a circle with two "Whiskers"
+# the whiskers are two circles that are attached to the robot
+# The whiskers center points cannot be outside of the robot's circle
+# The robot also has a vaccum that is a rectangle that is under the robot
+# The vaccum's width cannot be larger than the robot's diameter
+
+
+# The whiskers have a diameter, and a position relative to the robot they are attached to
+class Whisker:
+    # Constructor
+    # x_pos: The x position of the whisker
+    # y_pos: The y position of the whisker
+    # diameter: The diameter of the whisker
+    def __init__(self, x_pos, y_pos, diameter):
+        self.x_pos = x_pos
+        self.y_pos = y_pos
+        self.diameter = diameter
+
+
+class Robot:
+    # Constructor
+    # x_pos: The x position of the robot
+    # y_pos: The y position of the robot
+    # diameter: The diameter of the robot in inches
+    # whisker_length: The length of the whiskers on the robot
+    # vaccum_width: The width of the vaccum on the robot
+    def __init__(
+        self,
+        x_pos,
+        y_pos,
+        diameter=12.8,
+        whisker_length=13.5,
+        vaccum_width=5.8,
+    ):
+        self.x_pos = x_pos
+        self.y_pos = y_pos
+        self.diameter = diameter
+        self.whisker_length = whisker_length
+        self.whiskers = []
+        self.vaccum_width = vaccum_width
+        self.is_valid = True
+
+        # Create the whiskers
+        self.whiskers.append(
+            Whisker(
+                x_pos + self.diameter / 2,
+                y_pos + self.diameter / 2,
+                self.whisker_length,
+            )
+        )
+        self.whiskers.append(
+            Whisker(
+                x_pos - self.diameter / 2,
+                y_pos + self.diameter / 2,
+                self.whisker_length,
+            )
+        )
+
+    # Validate the robot
+    # @return: True if the robot is valid, False otherwise
+    def validate(self):
+        # Is the vaccum width larger than the robot's diameter?
+        if self.vaccum_width > self.diameter:
+            self.is_valid = False
+            return False
+
+        # Are the centers of the whiskers outside of the robot?
+        for whisker in self.whiskers:
```

Comment:
I'm increasingly certain this Validate function is all wrong - there's a check against a *diamond* of all things first, the loop's being broken if one whisker fits, the fallthru condition is a return. It'll never get to the second whisker as far as I can tell.

Reply:
I am Not sure what you mean by diamond, but you are right about the return issue. Breaking as soon as it is false is okay, but right now it never returns true. Will fix